heroku/cli
# feat(cli): Add Fir support to 'pipelines:diff'

## Description

Here we add Fir pipelines support to command `pipelines:diff` requiring to read the Github commit SHA from the release OCI image, instead of the slug used for Cedar apps.

It was required to introduce some new interfaces based on the Fir types on a shared library and that made necessary to update other usages of the same functions in other `pipelines` topic commands, but changes are minimal.

## Testing

- Fetch this branch, run `yarn` and `yarn build`.
- Run `heroku pipelines:diff -a particleboard-staging`. There shouldn't be any errors and it should be up to date with `particleboard-production`.
- Repeat using `./bin/run pipelines:diff -a particleboard-staging`. The output should be exactly the same, no errors.
- Run `heroku pipelines:diff -a my-ruby-gsg-staging`. It errors out stating that no release was found, because of the lack of support for Fir apps.
- Repeat using `./bin/run pipelines:diff -a my-ruby-gsg-staging`. It shouldn't error out, showing that `my-ruby-gsg-staging` is 1 commit ahead of `my-ruby-gsg-production`.

## SOC2
GUS WI: [W-16679417](https://gus.lightning.force.com/a07EE000020liNkYAI)
